### ISSUE TITLE
Error | Could not find plugin "proposal-numeric-separator" | Generating ES5 bundles for differential loading

### ISSUE THREAD

**Issue by vegarda** [angular/angular-cli#17262]
# 🐞 Bug report

### Command (mark with an `x`)

- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Is this a regression?

 Yes, the previous version in which this bug was not present was: a few hours ago

### Description

 A clear and concise description of the problem...
Build doesn't work

## 🔬 Minimal Reproduction

<pre><code>
npm install
node --max-old-space-size=4096 ./node_modules/@angular/cli/bin/ng build --prod
</code></pre>

## 🔥 Exception or Error
<pre><code>

Generating ES5 bundles for differential loading...
An unhandled exception occurred: [BABEL] C:\angular\dist\common-es2015.b1722ef05b43f664abc2.js: Could not find plugin "proposal-numeric-separator". Ensure there is an entry in ./available-plugins.js for it. (While processing: "C:\\angular\\node_modules\\@angular-devkit\\build-angular\\node_modules\\@babel\\preset-env\\lib\\index.js")
See "C:\Users\Vegard\AppData\Local\Temp\ng-J7YoSB\angular-errors.log" for further details.
</code></pre>

## 🌍 Your Environment
<pre><code>
Angular CLI: 8.3.25
Node: 12.14.1
OS: win32 x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic
... platform-webworker, platform-webworker-dynamic, router
... service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.25
@angular-devkit/build-angular     0.803.25
@angular-devkit/build-optimizer   0.803.25
@angular-devkit/build-webpack     0.803.25
@angular-devkit/core              8.3.25
@angular-devkit/schematics        8.3.25
@angular/cdk                      8.2.3
@angular/cli                      8.3.25
@angular/material                 8.2.3
@angular/pwa                      0.803.25
@ngtools/webpack                  8.3.25
@schematics/angular               8.3.25
@schematics/update                0.803.25
rxjs                              6.5.4
typescript                        3.5.3
webpack                           4.39.2

</code></pre>

**Anything else relevant?**

https://github.com/facebook/create-react-app/issues/8680

**Comment by keitoaino** [comment]
Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. Tis is the only workaround which helped me.

**Comment by dex252** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

This is don't work for me...

**Comment by azhukov87** [comment]
Hi, all
I have the issue as well, but i managed to find single working set of package(s)

"@angular-devkit/build-angular": "^0.803.23",
    "@angular/cli": "~8.3.22",`
`

I believe the you need to use lower version of @angular-devkit/build-angular rather then 0.803.25

**Comment by vegarda** [comment]
I was able to build by disabling generating ES5 bundles for differential loading by setting 

<pre><code>"browserslist": [
    "last 2 Chrome versions"
]</code></pre>

in packages.json, and 

<pre><code>"es5BrowserSupport": false</code></pre>

in angular.json

**Comment by mattcwebster** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

This also worked for me.  I was getting this error when building an image for an Angular 8 Universal app with docker.  I also followed the steps listed in this react thread to use npm resolutions.  https://github.com/facebook/create-react-app/issues/8680.  No sure if the resolutions step is needed because it wasn't until I added @babel/compat-data to my devDependencies that the issue was resolved.

**Comment by judedaryl** [comment]
> Try to install "@babel/compat-data": "~7.8.0" to devDependencies. This is the only workaround which helped me.

This also worked for me. I want to add that the error didn't show up locally, but only when i used Azure Devops. I think DevOps bumps up the babel versions

**Comment by OyebisiJemil** [comment]
> "@babel/compat-data": "~7.8.0"

this(npm i @babel/compat-data@7.8.0) works for me! thanks!

**Comment by SergeyMell** [comment]
> "@babel/compat-data": "~7.8.0"
This also worked for me. Be sure to use exactly this version. Pay attention to the version. Firstly I automatically installed `"@babel/compat-data": "^7.9.0"` - this didn't work

**Comment by keitoaino** [comment]
@judedaryl this could be also related to `package-lock.json` - if it's ignored (eg using `npm i` instead of `npm ci`) then remote CI could be affected by this issue.

**Comment by farajfarook** [comment]
just installing `"@babel/compat-data": "7.8.0"` didn't work for me. I had to delete the `package-lock.json` file

> > Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.
> 
> This is don't work for me...

**Comment by raffienficiaud-pix4d** [comment]
Sticking to previous version of `build-angular` just do the work without any other change (and `"@babel/compat-data": "7.8.0"` did not work for me, with or without removing the lock)

```
"devDependencies": {
    "@angular-devkit/build-angular": "=0.803.22",
    ...
}
```

**Comment by morinellipy** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

This worked for me. Is this considered a temporary solution? Should we keep a watch on this and remove it later?

**Comment by zraees** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

Rocks!

**Comment by gs-nchakravarthi** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

Worked for me!! Do we need to remove it manually after the fix or will be handled in the internal devDependencies?

**Comment by keitoaino** [comment]
@morinellipy @gs-nchakravarthi yes, we need to keep this only until issue will be fixed. Then it will be needed to remove `@babel/compat-data` dependency from host website and update `@angular-devkit/build-angular` (see https://github.com/angular/angular-cli/pull/17267)

**Comment by gustavohks2** [comment]
"@babel/compat-data": "~7.8.0" suggestion from the comment above solved the problem. I believe it will be fixed soon since the issue has been reported yesterday.

**Comment by vajidaliji** [comment]
> > "@babel/compat-data": "~7.8.0"
> 
> this(npm i @babel/compat-data@7.8.0) works for me! thanks @keitoaino

Thanks men this is also work for me

**Comment by Roman-Lo** [comment]
> > Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.
> 
> This also worked for me. I was getting this error when building an image for an Angular 8 Universal app with docker. I also followed the steps listed in this react thread to use npm resolutions. [facebook/create-react-app#8680](https://github.com/facebook/create-react-app/issues/8680). Not sure if the resolutions step is needed because it wasn't until I added @babel/compat-data to my devDependencies that the issue was resolved. I would first just try adding @babel/compat-data to devDependencies.

I'm using nrwl workspace with angular@^8.2.0(the angular cli version is 8.3.14 in the yarn.lock) and I having the same issue.

I've tried all the work arounds under this thread and none of them works for me, not until I tried adding `resolutions` in my package.json file. 
```json
"resolutions": {
    "@babel/preset-env": "^7.8.7"
},
```
See: [create-react-app Issue #8860](https://github.com/facebook/create-react-app/issues/8680#issuecomment-601851525)

I hope this will help someone who still having issue after trying all above, like me. And I'm looking forward for the fix.

**Comment by lchiesa2000** [comment]
The problem is preseent also with angular 8
[INFO] Generating ES5 bundles for differential loading...
[INFO] An unhandled exception occurred: [BABEL] /var/lib/jenkins/workspace/dgkoro1_0/dg-suite-client/dist/dg-suite/polyfills-es5.42ee81f782be21ac5c8b.js: Could not find plugin "proposal-numeric-separator". Ensure there is an entry in ./available-plugins.js for it. (While processing: "/var/lib/jenkins/workspace/dgkoro1_0/dg-suite-client/node_modules/@babel/preset-env/lib/index.js")
[INFO] See "/tmp/ng-QZa8t9/angular-errors.log" for further details.
[INFO] npm ERR! code ELIFECYCLE
[INFO] npm ERR! syscall spawn
[INFO] npm ERR! file sh
[INFO] npm ERR! errno ENOENT
[INFO] npm ERR! dg-suite@0.0.0 build-svil: `ng build --configuration=svil && node usestrictfixer.js`
[INFO] npm ERR! spawn ENOENT
[INFO] npm ERR! 
[INFO] npm ERR! Failed at the dg-suite@0.0.0 build-svil script.
[INFO] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
[INFO] 
[INFO] npm ERR! A complete log of this run can be found in:
[INFO] npm ERR!     /var/lib/jenkins/.npm/_logs/2020-03-23T07_48_51_372Z-debug.log

**Comment by lakshitha1629** [comment]
delete node_modules and package-lock.json

add "resolutions": { "@babel/preset-env": "^7.8.7" } to package.json

npm install npm-force-resolutions --save-dev
npm install
npx npm-force-resolutions
npm install again
ionic build

**Comment by alexnoise79** [comment]
CHANGING
`"@angular-devkit/build-angular": "^0.803.25" `
TO
`"@angular-devkit/build-angular": "0.803.23"`

worked for me

**Comment by mohanramphp** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

Thank god. This works. Angular Team please help in fixing this issue from angular cli end.

**Comment by splincode** [comment]
works

```ts
"@babel/compat-data": "7.8.0"
```

no works

```ts
"@babel/compat-data": "7.9.0"
```

**Comment by kkilton** [comment]
Verified that it was a Babel bug introduced 3 days ago.
https://github.com/babel/babel/pull/11201
https://www.npmjs.com/package/@babel/preset-env
Lock down the versions of Babel’s packages using the tilde instead of the caret:
`"dependencies": {
    "@babel/preset-env": "~7.8.7",
    "@babel/compat-data": "~7.8.6",`

**Comment by stchln** [comment]
Hi kkilton, may I ask you how to do this lockdown? Is it equivalent procedure than the one given by Lakshitha Perera but for preset-env and compat-data ?

**Comment by kkilton** [comment]
> Hi kkilton, may I ask you how to do this lockdown? Is it equivalent procedure than the one given by Lakshitha Perera but for preset-env and compat-data ?

stchln, I didn't need to do all the extra steps that Lakshitha did (although I tried and it didn't work for me) I just updated the dependencies in the package.config. The tilde's are what's important. The caret will allow the versions to go up to minor 7.9.999 (which is where the problem is, minor version 9) where the tilde only allows the patch version to increase up to 7.8.999... You could also remove the tilde and caret completely until Babel gets a fix deployed. Hope this helps.

**Comment by stchln** [comment]
Will try, thanks a lot

**Comment by Ruslancic** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

This procedure, break the compilation of es5 files in ionic for old webview system and older phone!

**Comment by NewtonMan** [comment]
WORKARROUND
On package.json add 
```
  "resolutions": {
    "@babel/preset-env": "^7.8.7",
    "@babel/compat-data": "~7.8.0"
  }
```
run `npx npm-force-resolutions`
then run `npm install` or `yarn`
then run `npm build` or `yarn build`

**Comment by jazarie2** [comment]
> WORKARROUND
> On package.json add
> 
> ```
>   "resolutions": {
>     "@babel/preset-env": "^7.8.7",
>     "@babel/compat-data": "~7.8.0"
>   }
> ```
> 
> run `npx npm-force-resolutions`
> then run `npm install` or `yarn`
> then run `npm build` or `yarn build`

Tried this and work on first try

**Comment by stchln** [comment]
This is not working for me, I still have the painful error on @babel/plugin-proposal-numeric-separator while in my package-lock.json I see compat-data 7.8.6 and preset-env 7.8.7

npm ls @babel/preset-env gives
visual@ c:...
— @babel/preset-env@7.8.7

npm ls @babel/compat-data 
visual@ ...
+— @babel/compat-data@7.8.6
ˋ— @babel/preset-env@7.8.7
   +— @babel/compat-data@7.8.6 deduped 
   ´— @babel/helper-compilation-target@7.8.7
     ´@babel/compat-data@7.8.6

**Comment by stchln** [comment]
@jazarie2 just tried what you suggest and I can go up to npm build, thanks but then when, as sampleRHTMLVisual demo I am trying to build, I am requested to have @babel/plugin-proposal-numeric-separator when launching pbiviz package...

**Comment by stchln** [comment]
And If I do a npm install —save-dev @babel/plugin-proposal-numeric-separator I get a warning that the plugin required a peer of @babel/core@^7.0.0-0 but none is installed

**Comment by kkilton** [comment]
@stchln can you try and delete your node_module/@babel folder and then run `npm install`? I went down the route of trying to install the proposal-numeric-separator but ended up reverting that and didn't need it in the end. Your package-lock.json looks fine from the ls cmd but maybe try and delete it and re-gen it with `npm install --package-lock?`

**Comment by MustafaMohamed2018** [comment]
> working
> 
> ```ts
> "@babel/compat-data": "7.8.0"
> ```
> 
> not working
> 
> ```ts
> "@babel/compat-data": "7.9.0"
> ```

that worked for me thanks

**Comment by datazen** [comment]
> WORKARROUND
> On package.json add
> 
> ```
>   "resolutions": {
>     "@babel/preset-env": "^7.8.7",
>     "@babel/compat-data": "~7.8.0"
>   }
> ```
> 
> run `npx npm-force-resolutions`
> then run `npm install` or `yarn`
> then run `npm build` or `yarn build`

Tried this and also worked on first try

**Comment by lopezbertoni** [comment]
Same here 

> > working
> > ```ts
> > "@babel/compat-data": "7.8.0"
> > ```
> > 
> > 
> > not working
> > ```ts
> > "@babel/compat-data": "7.9.0"
> > ```
> 
> that worked for me thanks

**Comment by burdokat** [comment]
> 
> 
> CHANGING
> `"@angular-devkit/build-angular": "^0.803.25" `
> TO
> `"@angular-devkit/build-angular": "0.803.23"`
> 
> worked for me

This worked for me. No problem with angular ng build.

**Comment by willr79** [comment]
> WORKARROUND
> On package.json add
> 
> ```
>   "resolutions": {
>     "@babel/preset-env": "^7.8.7",
>     "@babel/compat-data": "~7.8.0"
>   }
> ```
> 
> run `npx npm-force-resolutions`
> then run `npm install` or `yarn`
> then run `npm build` or `yarn build`

This also worked for me on the first try.  Thank you for the solution!

**Comment by fritzwf** [comment]
Tried all the workarounds and none worked for me.

**Comment by devexpert1** [comment]
Try this, this works:  npm i @babel/compat-data@7.8.0

**Comment by stchln** [comment]
DONE compiled pbiviz after applying the « resolutions » from datazen + manual installation of @babel/plugin-proposal-numeric-separator 
Thanks all

**Comment by WahabShah23** [comment]
> WORKARROUND
> On package.json add
> 
> ```
>   "resolutions": {
>     "@babel/preset-env": "^7.8.7",
>     "@babel/compat-data": "~7.8.0"
>   }
> ```
> 
> run `npx npm-force-resolutions`
> then run `npm install` or `yarn`
> then run `npm build` or `yarn build`

This worked for me as well. Tried all others but this was the successfully executed. Have been banging my head from past 3-4 hours before landed here. Thank you.

**Comment by WahabShah23** [comment]
> Try this, this works: npm i @babel/compat-data@7.8.0

This worked for me as well. Tried all others but this was the one successfully executed. Have been banging my head from past 6-7 hours before landed here. Thank you.

**Comment by astaev** [comment]
Manually referencing the numeric-separator package into the @babel/preset-env/lib/available-plugins.js fixes the error.

**Comment by vigan-abd** [comment]
> Hi, all
> @vegarda, @dex252
> I have the issue as well, but i managed to find single working set of package(s)
> 
> "@angular-devkit/build-angular": "^0.803.23",
> "@angular/cli": "~8.3.22",` `
> 
> I believe the you need to use lower version of @angular-devkit/build-angular rather then 0.803.25

Forcing version of "@angular-devkit/build-angular" to 0.803.22 fixed the issue also in my env, "@babel/compat-data": "~7.8.0" didn't work

**Comment by okandavut** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

It's also worked for me thanks!

**Comment by pixael** [comment]
I only had to do:
`npm i @babel/compat-data@7.8.0`
works like a charm. Thanks to @keitoaino.

**Comment by pehfernandesl** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

This worked to me.

Angular CLI: 8.3.25
Node: 12.16.1
OS: linux x64
Angular: 8.2.14

@angular-devkit/architect         0.803.25
@angular-devkit/build-angular     0.803.25
@angular-devkit/build-optimizer   0.803.25
@angular-devkit/build-webpack     0.803.25
@angular-devkit/core              8.3.25
@angular-devkit/schematics        8.3.25
@angular/cli                      8.3.25
@ngtools/webpack                  8.3.25
@schematics/angular               8.3.25
@schematics/update                0.803.25
rxjs                              6.4.0
typescript                        3.5.3
webpack                           4.39.2

**Comment by unnhao** [comment]
I remove whole node_modules which installed with ng new, and run npm install again.
```
Angular CLI: 8.3.25
Node: 10.17.0
OS: win32 x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms     
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.25
@angular-devkit/build-angular     0.803.25
@angular-devkit/build-optimizer   0.803.25
@angular-devkit/build-webpack     0.803.25
@angular-devkit/core              8.3.25
@angular-devkit/schematics        8.3.25
@angular/cli                      8.3.25
@ngtools/webpack                  8.3.25
@schematics/angular               8.3.25
@schematics/update                0.803.25
rxjs                              6.4.0
typescript                        3.5.3
webpack                           4.39.2
```

**Comment by adiediekdo** [comment]
> > "@babel/compat-data": "~7.8.0"
> 
> this(npm i @babel/compat-data@7.8.0) works for me! thanks @keitoaino

work for me to

**Comment by nkch8054** [comment]
No need to install anything else 
Only had to do :
```
rm -rf node_modules dist 
npm install
```
It works

**Comment by yusufali19** [comment]
> > WORKARROUND
> > On package.json add
> > ```
> >   "resolutions": {
> >     "@babel/preset-env": "^7.8.7",
> >     "@babel/compat-data": "~7.8.0"
> >   }
> > ```
> > 
> > 
> > run `npx npm-force-resolutions`
> > then run `npm install` or `yarn`
> > then run `npm build` or `yarn build`
> 
> Tried this and work on first try

**Comment by barryguda-1** [comment]
"devDependencies": { 
       "babel/compat-data": "7.8.0",
}
This worked for me on first trial

**Comment by yanestevesufjf** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

works for me too.

**Comment by wzoccarato** [comment]
"devDependencies": {
"babel/compat-data": "7.8.0",
}

This worked for me too

**Comment by pimol** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

For me too, but only after I did `ionic repair`
P.

**Comment by pimol** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

For me too, but only after I run `ionic repair`

P.

**Comment by ozzpy** [comment]
work for me adding  "@babel/compat-data": "7.8.0" and delete lock file, then npm i, after this I was able to run the app.

**Comment by cat0rgi0** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

Worked

**Comment by rk-2903** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.
Do:-
npm i @babel/compat-data@7.8.0

**Comment by alexlarraga1** [comment]
> Sticking to previous version of `build-angular` just do the work without any other change (and `"@babel/compat-data": "7.8.0"` did not work for me, with or without removing the lock)
> 
> ```
> "devDependencies": {
>     "@angular-devkit/build-angular": "=0.803.22",
>     ...
> }
> ```

This works for me

**Comment by id1945** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

Thanks you! :+1:  :1st_place_medal:  :1st_place_medal:

**Comment by robertrypula** [comment]
I'm working with Yarn 1.22.4 and Angular 8.2.14 on Windows 10. Problem started to occur when I installed ngx-markdown 9.0.0. I had some TypeScript related issue so I downgraded ngx-markdown to 8.2.2. After that I started to get proposal-numeric-separator errors. In my case adding ONLY this...
```json
"resolutions": {
    "@babel/preset-env": "^7.8.7"
},
```
...to the package.json worked. Probably I also removed yarn-lock and node_modules before installing packages once more time (I spent "magic" 1.5h of frustration...). At the same time I "destroyed" my Ubuntu 18.04 VPS . It was working fine on commits before installing ngx-markdown. Nothing was helping, even resetting repository to the commit from the past ;) Looks like when it's broken it's remains broken no matter what you clean or reinstall :P

**Comment by SaraSultanLearn** [comment]
with me none of previous solutions solved my issue:
Just Update Angular-cli in global and also update Node , npm
and create another new Project works fine :)

**Comment by phantomk** [comment]
> Sticking to previous version of `build-angular` just do the work without any other change (and `"@babel/compat-data": "7.8.0"` did not work for me, with or without removing the lock)
> 
> ```
> "devDependencies": {
>     "@angular-devkit/build-angular": "=0.803.22",
>     ...
> }
> ```

worked for me, mac and docker(centos)

**Comment by pcaserta** [comment]
Make sure you hit  npm install  before rebuilding your app, or the actual version won't be installed

**Comment by virtualvishwam** [comment]
> I was able to build by disabling generating ES5 bundles for differential loading by setting
> 
> ```
> "browserslist": [
>     "last 2 Chrome versions"
> ]
> ```
> 
> in packages.json, and
> 
> ```
> "es5BrowserSupport": false
> ```
> 
> in angular.json

Where exactly to add these inside angular.json and package.json?
I am a bit new to all this.

Also installing "@babel/compat-data": "7.8.0" did not work for me.

I am using Ionic App

**Comment by mirogrenda** [comment]
The following worked for me too:

1. ADD "@babel/compat-data": "7.8.0" to package.json
2. CLEAN node_modules
3. DELETE package-lock.json
4. RUN npm install
5. RUN ng build

**Code example (from package.json):**
`{
  "scripts": {
    // no change here
  },
  "dependencies": {
    // no change here
  },
  "devDependencies": {
    "@babel/compat-data": "7.8.0" // added
  }
}`

**Comment by Fluuub** [comment]
All above was not working for me.
What I did as a TEMPORARY solution (Hence ugly):
npm install install @babel/plugin-proposal-numeric-separator --save-dev

Then changed file (And that is what is ugly ;p):
node_modules\@babel\preset-env\lib\available-lugins.js

By adding:
var _pluginProposalNumericSeparator = _interopRequireDefault(require("@babel/plugin-proposal-numeric-separator"));

And at the bottom, in the _default array, adding:
  "proposal-numeric-separator": _pluginProposalNumericSeparator

**Comment by vivek437** [comment]
"@babel/core": "^7.7.4",

> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

This did not work for me... but this did..

"devDependencies": {
"babel/compat-data": "7.8.0",
}

**Comment by vivek437** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

this did not work for me as well.. but this did..
  

> "devDependencies": {
>       "@babel/core": "^7.7.4",
> }

**Comment by sethbergman** [comment]
### CICD Use Case

- I'm building an application on a GitLab instance and deploying to kubernetes. 
- The project is `@angular/cli` version 7.x.x and I want to use Node.js 12 in my Dockerfile.

I clone the repo to remove the package lock and the `node_modules` file / folder.

I add [this suggestion from above](https://github.com/angular/angular-cli/issues/17262#issuecomment-604847854) to my `package.json` then do a quick install to generate a new `package-lock.json`, and then I modify my Dockerfile. This particular project doesn't require me to compile so I'm doing everything in the Dockerfile.

```dockerfile
FROM  custom-registry/nodejs-devops:12-lts_v2.0.0 as node
WORKDIR /app
COPY . .
#################################################
RUN npm install && npx npm-force-resolutions
##################################################
ENV NODE_ENV=production
RUN npm run build

FROM custom-registry/nginx-devops:1.16-lts_v2.0.0
EXPOSE 8080
COPY --from=node /app/dist /usr/share/nginx/html
```

**Comment by SemicolonStruggles** [comment]
Can confirm that manually adding "@babel/compat-data": "~7.8.0" to the devDependencies works. 

You need to explicitly tell it to use version 7.8.0 (using **~** rather than **^**) since any newer versions will not work.

**Comment by clydin** [comment]
For anyone experiencing this issue, Angular CLI 8.3.26 has been released with an update to the babel versions.  No workarounds should be necessary after an update.  When updating, please ensure that the `@angular-devkit/build-angular` package was also update to 0.803.26.  This package contains the majority of the build logic for the CLI.

If anyone is still encountering issues after an update, please open a new issue with details regarding your environment and project versions.

**Comment by Seshirantha** [comment]
`npm i @babel/compat-data@7.8.0` worked for me. Check the followings as well.

Results of `ng --version`
````
     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/


Angular CLI: 8.3.25
Node: 10.16.3
OS: win32 x64
Angular: 8.2.14
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.803.25
@angular-devkit/build-angular     0.803.25
@angular-devkit/build-optimizer   0.803.25
@angular-devkit/build-webpack     0.803.25
@angular-devkit/core              8.3.25
@angular-devkit/schematics        8.3.25
@angular/cli                      8.3.25
@ngtools/webpack                  8.3.25
@schematics/angular               8.3.25
@schematics/update                0.803.25
rxjs                              6.4.0
typescript                        3.5.3
webpack                           4.39.2````

**Comment by UsmanAhmad888** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

works for me too (Y) thanks

**Comment by devsandro** [comment]
> Try to install `"@babel/compat-data": "~7.8.0"` to `devDependencies`. This is the only workaround which helped me.

npm i @babel/compat-data@7.8.0 this worked fine. thanks

**Comment by Johannes-LAMK** [comment]
I had the same problem, when building angular app. I just updated my packages and then it worked.

**Comment by anantkumar323** [comment]
delete node_modules and package-lock.json
add “resolutions”: { “@babel/preset-env”: “^7.8.7” }, to package.json

**Comment by kael1706** [comment]
I solved this problem as follows:
```
"devDependencies": {
    "@angular-devkit/architect": "~0.801.2",
    ...
```
in package.json.
then I did `npm install` to apply the changes.

**Comment by gokul123** [comment]
got this error while running ng build. 
Solution
1. add "@babel/compat-data": "~7.8.0" to devDependencies. 
2. npm install

**Comment by cvele89** [comment]
> 
> 
> got this error while running ng build.
> Solution
> 
>     1. add "@babel/compat-data": "~7.8.0" to devDependencies.
> 
>     2. npm install

Helped me as well.

**Comment by dominic-ks** [comment]
> > got this error while running ng build.
> > Solution
> > ```
> > 1. add "@babel/compat-data": "~7.8.0" to devDependencies.
> > 
> > 2. npm install
> > ```
> 
> Helped me as well.

Same for me, started experiencing this error after updating an Ionic 4 app to Ionic 5 and building with Ionic Pro.

**Comment by ZakoMthiya** [comment]
> Can confirm that manually adding "@babel/compat-data": "~7.8.0" to the devDependencies works.
> 
> You need to explicitly tell it to use version 7.8.0 (using **~** rather than **^**) since any newer versions will not work.

Thank you. This worked for me as well

**Comment by harshkr101** [comment]
` npm i @babel/compat-data@7.8.0`
 worked for me

**Comment by bullwrinkle** [comment]
I've tried all the various suggestions listed above and none of them are working for m.

  "devDependencies": {
    "@babel/compat-data": "^7.8.0",
    "@babel/core": "^7.10.3",
    "@babel/runtime": "^7.10.3",
     ..
}

$ rm -rf node_modules
$ yarn cache clean
$ yarn

[19:17:49] Error in plugin "gulp-babel"
Message:
    [BABEL] /Users/normanklein/Library/Caches/Yarn/v6/.tmp/dab8deaa179dd66562dd889ce1b3b406.27a5ca00f41160791f80b8ed70bf3ce5afad5c0b.prepare/src/Analytics.js: Could not find plugin "proposal-numeric-separator". Ensure there is an entry in ./available-plugins.js for it. (While processing: "/Users/normanklein/Library/Caches/Yarn/v6/.tmp/dab8deaa179dd66562dd889ce1b3b406.27a5ca00f41160791f80b8ed70bf3ce5afad5c0b.prepare/node_modules/@babel/preset-env/lib/index.js")
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! parse@2.11.0 gulp: `gulp "compile"`
npm ERR! Exit status 1
